test/testdata/infer/takes_no_block_inject.rb

```diff
@@ -1,4 +0,0 @@
-# typed: strict
-
-foo = T::Array[Integer].new
-foo.inject(nil) {|sum, x| sum ? sum + x : x}
```

Comment:
This was added by @jez on [[sorbet/pr/4063] Fix some RBIs that actually take blocks](https://github.com/sorbet/sorbet/pull/4063/)

I'm not entirely sure we do want it, though... now that the signature is tighter, the type error that tells you that `sum` is always `nil` initially will signal the author that they don't need to do this... It would make more sense to just write `foo.inject {|sum, x| sum ? sum + x : x}` in a case like this 🤷🏽 

Note that we didn't have the same test for `reduce` any way, and those are aliases

Reply:
The type error that Sorbet produces on this snippet is the biggest source of errors in Stripe's codebase, and the reason why I hadn't gone forward with this approach in my previous attempt.

> in most cases, too!

The history here is that Stripe's codebase had an `Array#sum` monkeypatch which clobbered the built-in `Array#sum` method, and that caused problems so we backed it out by inlining a bunch of these `.inject` calls all over the codebase.